src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -190,6 +191,52 @@ impl<'a, 'tcx> LinkCollector<'a, 'tcx> {
         }
     }
 
+    fn resolve_primitive_associated_item(
+        &self,
+        prim_ty: hir::PrimTy,
+        prim: Res,
```

Comment:
Why does this take both? If you need the full `Res` I'd expect this to generate the `prim_ty` from the Res.

Reply:
It didn't occur to me that I can "just" repack `prim_ty`. Thanks. While this is the exact opposite of what you wrote, I think the pattern matching should be avoided for the Res -> PrimTy conv.

Reply:
It's a shame I can't somehow tell in code that "I know this must be a Res::PrimTy. I still need the whole enum value, but I also want it's contents without pattern matching". Sooo, I kinda have to go the other way and wrap the `prim_ty` inside the function instead. I'm not so keen on adding an `unreachable!()` for pattern matching, or an unnecessary unwrap somewhere here.

Reply:
Yeah, I ran into the same issue a few days ago: https://discordapp.com/channels/273534239310479360/274215136414400513/761661592898633748. This would be partially addressed by https://github.com/rust-lang/rfcs/pull/2593 I think.